### ISSUE TITLE
Credo.Check.Consistency.UnusedVariableNames fails with error

### ISSUE THREAD

**Issue by retgoat** [rrrene/credo#717]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..."
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.2.0-dev
* Erlang/Elixir version (`elixir -v`): 
  * Erlang/OTP 22 [erts-10.4.3] [source] [64-bit] [smp:8:8][ds:8:8:10] [async-threads:1] [hipe] [dtrace]
   * Elixir 1.9.0 (compiled with Erlang/OTP 22)
* Operating system: MacOS Catalina 10.15.1 (19B88)

### What were you trying to do?
Enable `{Credo.Check.Consistency.UnusedVariableNames, []}`

### Expected outcome

Check for unused variables

### Actual outcome
```bash
** (EXIT from #PID<0.92.0>) an exception was raised:
    ** (Protocol.UndefinedError) protocol Enumerable not implemented for nil of type Atom
        (elixir) lib/enum.ex:1: Enumerable.impl_for!/1
        (elixir) lib/enum.ex:141: Enumerable.reduce/3
        (elixir) lib/enum.ex:3023: Enum.reduce/3
        lib/credo/check/consistency/unused_variable_names/collector.ex:28: Credo.Check.Consistency.UnusedVariableNames.Collector.traverse/3
        (elixir) lib/macro.ex:290: anonymous fn/4 in Macro.do_traverse_args/4
        (elixir) lib/enum.ex:1440: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
        (elixir) lib/enum.ex:1440: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
        (elixir) lib/macro.ex:256: Macro.do_traverse/4

```

**Comment by TheFirstAvenger** [comment]
@retgoat is there a reason you closed the PR that fixes this? I also see the same crash and the code change you proposed fixes the issue.

**Comment by retgoat** [comment]
@TheFirstAvenger yeah, that fixes. But code works in opposite direction. I mean credo pops up a warning to underscored variable instead of not used. So I got a lot of warnings for `_foo` variables

**Comment by TheFirstAvenger** [comment]
@retgoat And you are sure your codebase uses more `_model` than `_`? If you search across all files for "` _ `" and `_)`, does it find more instances of that than credo warnings for underscored variables?

**Comment by retgoat** [comment]
@TheFirstAvenger the issue is not there. Actually `_model` **uses** this convention as it named with underscore. But I intentionally left some unused variables without underscore in the beginning, and credo was not able to find these

**Comment by TheFirstAvenger** [comment]
@retgoat My understanding of this check is that it looks for consistency between using `_` or using `_unused_var`, warning if you use both instead of consistently using only one. It does care about `unused_var`, as that is a compiler warning.

**Comment by retgoat** [comment]
@TheFirstAvenger Maybe I read it incorrectly. Anyways, you applied the fix
